### PR TITLE
[CDEC-427] Move Arbitrary instances from ssc to ssc-test

### PR DESCRIPTION
**Pull request by intricate**
## Description

As it stands, the Arbitrary instances for data types in `ssc` all exist in `ssc` when they should actually be exposed from a separate package, `ssc-test`.

In addition to this, `Pos.Arbitrary.Ssc` was actually used within `Pos.Ssc.Worker`. It was used to [generate a random Opening](https://github.com/input-output-hk/cardano-sl/blob/develop/ssc/src/Pos/Ssc/Worker.hs#L235) in the event that we call [`onNewSlotOpening`](https://github.com/input-output-hk/cardano-sl/blob/develop/ssc/src/Pos/Ssc/Worker.hs#L206) while passing an instance of [`SscOpeningParams`](https://github.com/input-output-hk/cardano-sl/blob/develop/ssc/src/Pos/Ssc/Behavior.hs#L36) constructed with [`SscOpeningWrong`](https://github.com/input-output-hk/cardano-sl/blob/develop/ssc/src/Pos/Ssc/Behavior.hs#L39). This was changed not only because the `Arbitrary` instance deterministically generated an `Opening` (does anyone know whether there was any kind of security risk in this?), but also because `ssc` can't depend on `ssc-test` (circular dependency).

## Linked issue

[https://iohk.myjetbrains.com/youtrack/issue/CDEC-427](https://iohk.myjetbrains.com/youtrack/issue/CDEC-427)
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ssc/src/Pos/Ssc/Worker.hs [hunk 1]
```diff
@@ -202,6 +201,16 @@ onNewSlotCommitment pm slotId@SlotId {..} sendCommitment
         sscProcessOurMessage (sscProcessCommitment pm comm)
         sendOurData sendCommitment CommitmentMsg comm siEpoch 0
 
+-- | Generate a random Opening.
+randomOpening :: IO Opening
+randomOpening = snd <$> secureRandCommitmentAndOpening
+  where
+    secureRandCommitmentAndOpening = runSecureRandom $ do
+        t       <- randomNumberInRange 3 10
+        n       <- randomNumberInRange (t*2-1) (t*2)
```
Comment (erikd) [inline ssc/src/Pos/Ssc/Worker.hs]: That range does not look right :). It looks to me like the range only has two valid values.

Comment (erikd) [reply ssc/src/Pos/Ssc/Worker.hs]: I suspect that its actually supposed to be
```
       n       <- randomNumberInRange (2 ^ (t -1)) (2 ^ t)
```
but then the value of `t` should be restricted more than it is.

File: ssc/src/Pos/Ssc/Worker.hs [hunk 2]
```diff
@@ -232,7 +241,7 @@ onNewSlotOpening pm params SlotId {..} sendOpening
         mbOpen' <- case params of
             SscOpeningNone   -> pure Nothing
             SscOpeningNormal -> pure (Just open)
-            SscOpeningWrong  -> Just <$> liftIO (QC.generate QC.arbitrary)
+            SscOpeningWrong  -> Just <$> liftIO randomOpening
```
Comment (parsonsmatt) [inline ssc/src/Pos/Ssc/Worker.hs]: This is a lot better, certainly. But.. *why* are we generating a random opening here? This seems deeply weird to me.

Comment (intricate) [reply ssc/src/Pos/Ssc/Worker.hs]: I'm going to be digging into this today to really understand.

**Pull request merged**